Formula/libeatmydata.rb

```diff
@@ -0,0 +1,96 @@
+class Libeatmydata < Formula
+  desc "Because fsync() should be a no-op"
+  homepage "https://www.flamingspork.com/projects/libeatmydata/"
+  url "https://github.com/stewartsmith/libeatmydata/releases/download/v130/libeatmydata-130.tar.gz"
+  sha256 "48731cd7e612ff73fd6339378fbbff38dd3bcf6c243593b0d9773ca0051541c0"
+  license "GPL-3.0-or-later"
+  head "https://github.com/stewartsmith/libeatmydata.git", branch: "master"
+
+  depends_on "autoconf"         => :build
+  depends_on "autoconf-archive" => :build
+  depends_on "automake"         => :build
+  depends_on "libtool"          => :build
+
+  depends_on "coreutils" => :test
+  depends_on "grep"      => :test
+
+  on_linux do
+    depends_on "strace" => :test
+  end
+
+  def install
+    system "autoreconf", "-vfi"
+    system "./configure", *std_configure_args,
+                          "--disable-option-checking",
+                          "--disable-silent-rules"
+    system "make", "install"
+  end
+
+  test do
+    # 0 - PASS: Successful test
+    # 1 - FAIL: Result code missing
+    # 2 - FAIL: eatmydata command missing
+    # 3 - FAIL: sync command missing
+    # 4 - FAIL: Unable to export LD_PRELOAD
+    # 5 - FAIL: strace output missing
+    # 6 - FAIL: Library malfunction
+    # 7 - FAIL: Linux shared object missing
+    # 8 - FAIL: Missing macOS dynamic library
+    # 9 - FAIL: Unable to determine OS
+    (testpath/"test.sh").write <<~EOS
```

Comment:
This test is really opaque and unmaintainable. You'll need to rewrite this in our testing DSL, I'm afraid.

Even without the rewrite, it looks like the test only checks the existence of some files and some usage output, which we don't usually accept as tests for new formulae.

Reply:
Testing that the wrapper starts and that it installed is the only test possible on MacOS - it would require root access otherwise for any other tests.

I could simplify the Linux test otherwise to be just a single line or two - this test here is actually a simplified version of the upstream test.

Reply:
At least on the systems I can use, there is no strace on macOS - only dtruss which requires root.

Reply:
The main part of the test on Linux works by running strace on sync and ensuring that sync was not in the output - meaning that it is functioning correctly.... that's probably a one-liner alone.

Would that work?  I could write a MacOS test too, but we can't really use it without requesting sudo or being root.

Reply:
What does `eatmydata` do when you try to run it normally but not as root?

Reply:
Replied in comment, sorry ...

Reply:
Sure, that's fine. Note this thread isn't really resolved yet -- my initial comment regarding rewriting this in Ruby is not resolved.

Reply:
I'll try a Ruby test tonight.

I can easily write tests in ksh, C, FORTRAN, REXX, Tcl, Janet, and Go - but Ruby and Python are mostly alien to me but I'll do my best. :)

Reply:
Hopefully https://github.com/Homebrew/homebrew-core/pull/106536/commits/17b6dd2c0ae300792500a1d4eaa3e355de4c8c23 is better?